enterprise/cmd/cody-gateway/internal/actor/actor.go

```diff
@@ -41,6 +43,21 @@ type Actor struct {
 	Source Source `json:"-"`
 }
 
+func (a *Actor) GetID() string {
+	return a.ID
+}
+
+func (a *Actor) GetName() string {
+	if a.Name == "" {
+		return a.ID
+	}
+	return a.Name
+}
+
+func (a *Actor) GetSource() codygateway.ActorSource {
+	return codygateway.ActorSource(a.Source.Name())
```

Comment:
Should we make this nil-safe, i.e.

```suggestion
    if a.Source == nil {
        return "unknown"
    }
	return codygateway.ActorSource(a.Source.Name())
```

Reply:
Also should `Source.Name()` return type `codygateway.ActorSource`?

Reply:
Did some checks and the majority of call sites expect `a.Source.Name()` to return a `string`, I guess let's leave it for now until it becomes the minority?